### ISSUE TITLE
KeePassXC Browser Extension Not Working with Mullvad Browser on Fedora 41

### ISSUE THREAD

**Issue by mohammadRezaeian** [mullvad/mullvad-browser#367]
Hi, I installed the Mullvad browser from the official Mullvad repository (mullvad.net).

I followed all the steps mentioned in previous issues for setting up the KeePassXC browser extension, but the extension is still not working with the Mullvad browser on Fedora 41.

I recall reading in a previous issue that someone resolved this issue by using the KeePassXC package from the Arch repository and enabling Firefox support in KeePassXC. However, even after trying this approach, it still doesn't work on Fedora 41.

Could you please confirm if additional steps are required, or if this is a compatibility issue with Fedora 41?

thanks.

**Comment by ExperimentalCyborg** [comment]
The `native-messaging-hosts` thing KeePassXC is looking for doesn't seem to exist for Mullvad. Native Messaging might be disabled entirely, which would make the KeePassXC extension unable to work at all.

I also would like to be able to use the leading secure open source password manager with Mullvad... Much safer than having your credentials cross your clipboard every time or screwing around with auto-type.

**Comment by canon250** [comment]
I managed to fix that on Fedora 41 KDE by enabling Browser Integration on KeepassXC, then checking Firefox, going to ~/.mozilla, copied the folder `native-messaging-hosts` (with the .json file inside) to /usr/lib/mozilla and restarted Mullvad Browser.

Currently working fine on KeepassXC version 2.7.9 and Mullvad Browser version 14.0.7 both installed natively through DNF.

**Comment by mohammadRezaeian** [comment]
Hi, thank you for the work
it works fine. But sorry, could you please tell me why it works with that setup, but when we copied it to the ~/.Mullvad-browser, it doesn't work?

**Comment by canon250** [comment]
Hey @mohammadRezaeian,

To be honest, I’m not involved with the development team, nor am I a developer myself. I figured this out through trial and error and just wanted to share my solution. I’m sorry I can’t provide an accurate answer, but perhaps someone from the KeePassXC development team can shed some light on this?

**Comment by moodyknurd** [comment]
Hey,
Just ran into this issue myself today and I noticed this directory: `~/.local/share/torbrowser/tbb/x86_64/tor-browser_en-US/Browser/TorBrowser/Data/Browser/.mozilla/native-messaging-hosts`, could this be what we're looking for?

**Comment by canon250** [comment]
> Hey, Just ran into this issue myself today and I noticed this directory: `~/.local/share/torbrowser/tbb/x86_64/tor-browser_en-US/Browser/TorBrowser/Data/Browser/.mozilla/native-messaging-hosts`, could this be what we're looking for?

I'm not sure what are you referring to specifically, my guess is that you enabled on KeepassXC the toggle for Tor Browser and that file gets created there.

Do you still have the issue with the extension? If you follow my previous instructions it should work flawlessly.

**Comment by moodyknurd** [comment]
> > Hey, Just ran into this issue myself today and I noticed this directory: `~/.local/share/torbrowser/tbb/x86_64/tor-browser_en-US/Browser/TorBrowser/Data/Browser/.mozilla/native-messaging-hosts`, could this be what we're looking for?
> 
> I'm not sure what are you referring to specifically, my guess is that you enabled on KeepassXC the toggle for Tor Browser and that file gets created there.
> 
> Do you still have the issue with the extension? If you follow my previous instructions it should work flawlessly.

Right, sorry for the lack of clarity, your instructions did resolve my issue, thanks! I was just wondering out loud about the purpose of that directory, is all.

**Comment by canon250** [comment]
> > > Hey, Just ran into this issue myself today and I noticed this directory: `~/.local/share/torbrowser/tbb/x86_64/tor-browser_en-US/Browser/TorBrowser/Data/Browser/.mozilla/native-messaging-hosts`, could this be what we're looking for?
> > 
> > 
> > I'm not sure what are you referring to specifically, my guess is that you enabled on KeepassXC the toggle for Tor Browser and that file gets created there.
> > Do you still have the issue with the extension? If you follow my previous instructions it should work flawlessly.
> 
> Right, sorry for the lack of clarity, your instructions did resolve my issue, thanks! I was just wondering out loud about the purpose of that directory, is all.

Nice! I'm glad to hear (read) that. I hope the issue gets fixed though

**Comment by linktonull** [comment]
> Hi, thank you for the work it works fine. But sorry, could you please tell me why it works with that setup, but when we copied it to the ~/.Mullvad-browser, it doesn't work?

I created the directory `~/.mullvad-browser/.mozilla/native-messaging-hosts` and copied `~/.mozilla/native-messaging-hosts/org.keepassxc.keepassxc_browser.json` there.
Now, KeePassXC works well.

In KeePassXC, set the config location to `~/.mullvad-browser/.mozilla/native-messaging-hosts`, which should also work.


I'm using Debian 12 with KDE Plasma (Wayland). 
I installed Mullvad Browser from their [repository](https://repository.mullvad.net/deb/).

Hope this can help someone.
:)

**Comment by chris02000** [comment]
Hi all, same issue with Mullvad on Win 11 Pro is there a simple way to get this to work.
